experimental/packages/opentelemetry-sdk-metrics-base/src/view/View.ts

```diff
@@ -14,53 +14,107 @@
+export type ViewOptions = {
+  // View
+  /**
+   *  If not provided, the Instrument name will be used by default. This will be used as the name of the metrics stream.
+   */
+  name?: string,
+  /**
+   * If not provided, the Instrument description will be used by default.
+   */
+  description?: string,
+  /**
+   * If provided, the attributes that are not in the list will be ignored.
+   * If not provided, all the attribute keys will be used by default.
+   */
+  attributeKeys?: string[],
+  /**
+   * The {@link Aggregation} aggregation to be used.
+   */
+  aggregation?: Aggregation,
 
-export interface ViewStreamConfig {
+  // Instrument
+  /**
+   * The type of the Instrument(s).
+   */
+  instrumentType?: InstrumentType,
   /**
-   * the name of the resulting metric to generate, or null if the same as the instrument.
+   * Name of the Instrument(s) with wildcard support.
    */
-  name?: string;
+  instrumentName?: string,
+
+  // Meter
   /**
-   * the name of the resulting metric to generate, or null if the same as the instrument.
+   * The name of the Meter.
    */
-  description?: string;
+  meterName?: string;
   /**
-   * the aggregation used for this view.
+   * The version of the Meter.
    */
-  aggregation?: Aggregation;
+  meterVersion?: string;
   /**
-   * processor of attributes before performing aggregation.
+   * The schema URL of the Meter.
    */
-  attributesProcessor?: AttributesProcessor;
+  meterSchemaUrl?: string;
+};
+
+function isSelectorProvided(options: ViewOptions): boolean {
+  return (options.instrumentName == null &&
+    options.instrumentType == null &&
+    options.meterName == null &&
+    options.meterVersion == null &&
+    options.meterSchemaUrl == null);
 }
 
-/**
- * A View provides the flexibility to customize the metrics that are output by
- * the SDK. For example, the view can
- * - customize which Instruments are to be processed/ignored.
- * - customize the aggregation.
- * - customize which attribute(s) are to be reported as metrics dimension(s).
- * - add additional dimension(s) from the {@link Context}.
- */
 export class View {
   readonly name?: string;
   readonly description?: string;
   readonly aggregation: Aggregation;
   readonly attributesProcessor: AttributesProcessor;
+  readonly instrumentSelector: InstrumentSelector;
+  readonly meterSelector: MeterSelector;
 
-  /**
-   * Construct a metric view
-   *
-   * @param config how the result metric streams were configured
-   */
-  constructor(config?: ViewStreamConfig) {
-    this.name = config?.name;
-    this.description = config?.description;
-    this.aggregation = config?.aggregation ?? Aggregation.Default();
-    this.attributesProcessor = config?.attributesProcessor ?? AttributesProcessor.Noop();
+  constructor(viewOptions: ViewOptions) {
+    // If no criteria is provided, the SDK SHOULD treat it as an error.
+    // It is recommended that the SDK implementations fail fast.
+    if (isSelectorProvided(viewOptions)) {
```

Comment:
I'm confused by this assertion. Views without name specified should be allowed to not set instrument selection criteria. Like:

```js
new View({
 aggregation: DEFAULT_AGGREGATION
});
```

Reply:
I fully agree, I was quite surprised as well when I re-read the spec and found that this should not be allowed.

[Here](https://github.com/open-telemetry/opentelemetry-specification/blob/ca593258953811eb33197976b8e565299b874853/specification/metrics/sdk.md?plain=1#L180-L182) it states, in context of instrument selection criteria that:

> If no criteria is provided, the SDK SHOULD treat it as an error. It is
    recommended that the SDK implementations fail fast. Please refer to Error
    handling in OpenTelemetry for the general guidance.
    
That being said: it is not completely impossible to get the same outcome as before. It now requires providing a wildcard `instrumentName` like so:

```ts
new View({
 aggregation: DEFAULT_AGGREGATION,
 instrumentName: '*'
});
```

So even with this change to make it comply with the spec, the functionality is still retained, but I agree that this specific case is more confusing now than it was before. :confused: